client-v2/src/components/FrontsEdit/Front.tsx

```diff
@@ -193,14 +196,27 @@ class FrontComponent extends React.Component<FrontProps, FrontState> {
         <FrontContainer>
           <FrontContentContainer>
             <SectionContentMetaContainer>
+              <CollapseAllButton
```

Comment:
Could we rename this component to `CollapseExpandButton`? It's just we have an "expand all" button made out of a component called `CollapseAllButton`

Reply:
Absolutely, and thanks for spotting this -- as you say, this makes very little sense as it stands.

Reply:
I, personally, would go further with Anna’s suggestion and rename it (dynamically) and deduplicate it also in the UI. Less clutter. One button for both actions, just like all the rest of expand/collapse buttons. Also – less mouse movements. Less movements, that is, if not for a jump:

![jump](https://user-images.githubusercontent.com/6032869/60926602-f2969c80-a29e-11e9-8e53-1f2d76c8f17e.gif)

Reply:
@paperboyo this would be ideal, but the difficult is that 'expanded' and 'collapsed' are not discrete states.

For example -- we can be in a situation where some articles in some collections are open, and some are not. In this situation (which is I think the case the majority of the time for most users) both 'collapse all' and 'expand all' are valid actions.

It's only when everything is open, or everything is closed, that only one action applies -- and this situation is I think quite rare! But it might be that I've missed something here, let me know 👍

Reply:
@paperboyo -- also, re: collapsing, this is an artifact of the responsiveness of individual fronts, definitely not desirable as you point out, we should address it!